Add pdb (#547)
Opster/opensearch-k8s-operator: docs/userguide/main.md

  description: Sample tenant
```

### PodDisruptionBudget

The PDB (Pod Disruption Budget) is a Kubernetes resource that helps ensure the high availability of applications by defining the acceptable disruption level during maintenance or unexpected events.
It specifies the minimum number of pods that must remain available to maintain the desired level of service.
The PDB definition is unique for every nodePool.

> Please be aware that you must provide `MinAvailable` and `MaxUnavailable` configurations in order to use PDB feature.

Reviewer: ```suggestion
You must provide either `minAvailable` or `maxUnavailable` to configure PDB, but noth both.
```

Reviewer: This does not make clear to the user that only one of the fields can be set. Also: field names must start lowercase.

```yaml
apiVersion: opensearch.opster.io/v1
kind: OpenSearchCluster
...
spec:
   nodePools:
      - component: masters
        replicas: 3
        diskSize: "30Gi"
        pdb:
          enable: true

>           MinAvailable: 3
>           MaxUnavailable: 0 

Reviewer: ```suggestion
          minAvailable: 3
          maxUnavailable: 0 
```
Also: Example will not work with both fields being set

Reviewer: Comment was not addressed. Example cannot have both fields set, that would be invalid,

  description: Sample tenant
```

### PodDisruptionBudget

The PDB (Pod Disruption Budget) is a Kubernetes resource that helps ensure the high availability of applications by defining the acceptable disruption level during maintenance or unexpected events.
It specifies the minimum number of pods that must remain available to maintain the desired level of service.
The PDB definition is unique for every nodePool.
You must provide either `minAvailable` or `maxUnavailable` to configure PDB, but noth both.

```yaml
apiVersion: opensearch.opster.io/v1
kind: OpenSearchCluster
...
spec:
   nodePools:
      - component: masters
        replicas: 3
        diskSize: "30Gi"
        pdb:
          enable: true

>           MinAvailable: 3

Reviewer: must start lowercase

      - component: datas
        replicas: 7
        diskSize: "100Gi"
        pdb:
          enable: true

>           MaxUnavailable: 2 

Reviewer: must start lowercase
